codersdk/sse.go

```diff
@@ -0,0 +1,89 @@
+package codersdk
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+
+	"golang.org/x/xerrors"
+)
+
+type ServerSentEvent struct {
+	Type ServerSentEventType `json:"type"`
+	Data interface{}         `json:"data"`
+}
+
+type ServerSentEventType string
+
+const (
+	ServerSentEventTypePing  ServerSentEventType = "ping"
+	ServerSentEventTypeData  ServerSentEventType = "data"
+	ServerSentEventTypeError ServerSentEventType = "error"
+)
+
+func ServerSentEventReader(rc io.ReadCloser) func() (*ServerSentEvent, error) {
```

Comment:
As far as I can tell, none of these need to be exposed from the `codersdk`. I see `ping` is used as a const, but that seems entirely reasonable to just hardcode in both places since it's tested code anyways.

I'd prefer to keep as much private as we can in our SDK, because we want to limit breaking changes in the future to API consumers. This can obvs be fixed in a future PR!